flake.nix

```diff
@@ -388,6 +388,58 @@
               postUnpack = "sourceRoot=$sourceRoot/perl";
             });
 
+            passthru.python-bindings =
+              let
+                python = final.python3;
+              in python.pkgs.buildPythonPackage {
+                name = "nix";
+                format = "other";
+
+                src = self;
+                postUnpack = "sourceRoot=$sourceRoot/python";
+
+                nativeBuildInputs = [
+                  final.buildPackages.ninja
+                  (final.buildPackages.meson.override { python3 = python; })
+                  final.buildPackages.pkgconfig
+                  final.buildPackages.nix
+                ];
+
+                mesonFlags = [
+                  "-Dnix_binary=${final.lib.getBin final.nix}/bin/nix"
```

Comment:
This can be avoided by using `settings.nixBinDir + "/nix"` at runtime.

It would be great to avoid this entirely, but to do it properly would require #1221.

Reply:
Nice, that works!